keras_cv/models/segmentation/deeplab_v3/deeplab_v3.py

```diff
@@ -0,0 +1,208 @@
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.model.segmentation.DeepLabV3(
+        num_classes=1, backbone=backbone
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(training_dataset.take(10), epochs=3)
+    ```
+
+    Args:
+        num_classes: int, the number of classes for the detection model. Note
+            that the num_classes doesn't contain the background class, and the
+            classes from the data should be represented by integers with range
+            [0, num_classes).
+        backbone: Backbone network for the model. Should be a KerasCV model.
+        spatial_pyramid_pooling: Also known as Atrous Spatial Pyramid Pooling
+            (ASPP). Performs spatial pooling on different spatial levels in the
+            pyramid, with dilation.
+        segmentation_head: Optional `keras.Layer` that predict the segmentation
+            mask based on feature from backbone and feature from decoder.
+        segmentation_head_activation: Optional `str` or function, activation
+            functions between the `keras.layers.Conv2D` layers and the final
+            classification layer, defaults to `"relu"`.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        input_tensor: optional Keras tensor (i.e., output of `layers.Input()`)
+            to use as image input for the model.
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        backbone,
+        spatial_pyramid_pooling=None,
+        segmentation_head=None,
+        segmentation_head_activation="softmax",
+        dropout=0.0,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        **kwargs,
+    ):
+        if not isinstance(backbone, keras.layers.Layer) or not isinstance(
+            backbone, keras.Model
+        ):
+            raise ValueError(
+                "Argument `backbone` must be a `keras.layers.Layer` instance "
+                f" or `keras.Model`. Received instead "
+                f"backbone={backbone} (of type {type(backbone)})."
+            )
+
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+        if input_shape[0] is None and input_shape[1] is None:
+            input_shape = backbone.input_shape[1:]
+            inputs = keras.Input(tensor=input_tensor, shape=input_shape)
+
+        if input_shape[0] is None and input_shape[1] is None:
+            raise ValueError(
+                "Input shapes for both the backbone and DeepLabV3 cannot be "
+                "`None`. Received: input_shape={input_shape} and "
+                "backbone.input_shape={backbone.input_shape[1:]}"
+            )
+
+        encoder_outputs = self._make_deeplabv3_encoder(
```

Comment:
Let's inline these two methods in the `__init__` to match the style of our other models.